src/pathfinder.rs

```diff
@@ -0,0 +1,367 @@
+use lazy_static::lazy_static;
+use mut_static::MutStatic;
+use num::integer::sqrt;
+use pathfinding::prelude::astar;
+use serde::{Deserialize, Serialize};
+use std::borrow::Borrow;
+use std::hash::Hash;
+use std::num::ParseIntError;
+use std::sync::{Arc, RwLock};
+use std::usize::MAX;
+
+lazy_static! {
+    static ref NODES: MutStatic<Vec<NodeContainer>> = MutStatic::new();
```

Comment:
We should phase out lazy_static for once_cell, you should use that instead (we already have it included).